src/service/data_products/heatmap.py

```diff
@@ -114,3 +149,67 @@ async def get_column_info(
         doc = await get_collection_singleton_from_db(
             storage, self._colname_columns, collection_id, load_ver, bool(load_ver_override))
         return heatmap_models.Columns.construct(**remove_collection_keys(doc))
+
+    async def get_heatmap(
+        self,
+        r: Request,
+        collection_id: str = PATH_VALIDATOR_COLLECTION_ID,
+        limit: int = QUERY_VALIDATOR_LIMIT,
+        # TODO HEATMAP support count
+        # TODO HEATMAP support startfrom
+        load_ver_override: str | None = QUERY_VALIDATOR_LOAD_VERSION_OVERRIDE,
+        user: kb_auth.KBaseUser = Depends(_OPT_AUTH)
+    ) -> heatmap_models.HeatMap:
+        appstate = app_state.get_app_state(r)
+        storage = appstate.arangostorage
+        load_ver = await get_load_version(
+            storage, collection_id, self._id, load_ver_override, user)
+        return await self._query(  # may want a sort direction arg?
+            storage,
+            collection_id,
+            load_ver,
+            limit,
+            None,
+            False,
+            None,
+            False,
+        ) 
+        
+
+    async def _query(
+        # ew. too many args
+        self,
+        store: ArangoStorage,
+        collection_id: str,
+        load_ver: str,
+        limit: int,
+        internal_match_id: str | None,
+        match_mark: bool,
+        internal_selection_id: str | None,
+        selection_mark: bool,
+    ) -> heatmap_models.HeatMap:
+        data = []
+        await query_simple_collection_list(
+            store,
+            names.COLL_MICROTRAIT_DATA,
+            lambda doc: data.append(heatmap_models.HeatMapRow.parse_obj(
+                remove_collection_keys(doc))),
+            collection_id,
+            load_ver,
+            names.FLD_KBASE_ID,
+            sort_descending=False,
+            skip=0,
+            limit=limit,
+            internal_match_id=_prefix_id(_MATCH_ID_PREFIX, internal_match_id),
+            match_mark=match_mark,
+            internal_selection_id=_prefix_id(_SELECTION_ID_PREFIX, internal_selection_id),
+            selection_mark=selection_mark,    
+        )
+        vals = []
+        for r in data:  # lazy lazy lazy
+            vals += [c.val for c in r.cells]
```

Comment:
If I really wanted to be efficient I'd just run through the values and keep track of max / min myself rather than making a copy of the data. Anyway, pulling the data from arango is >> this calculation which is why I didn't bother with optimization here

Also to me those 2 loop comprehensions aren't very comprehensible